### DISCUSSION TITLE
[Question] Install macOS on SD card

### DISCUSSION THREAD

**Discussion by badstorm** [badstorm/surface-pro-7-opencore#54] [Ideas]
Someone has successfully managed to install macos on a SD card with the internal sd card reader?

**Comment by idalin** [comment]
nope. SD card can not be found in installer's disk util.
But I can install to SD card using this efi `链接:https://pan.baidu.com/s/13sVRRqZySKvUu5saWpeOvQ 
提取码:25i9`(from:https://www.bilibili.com/read/cv10042865)

**Comment by badstorm** [comment]
Thank you for your suggestion. Since the site is in Chinese and asks for verification, would you be able to turn the direct link to the efi.zip file?

**Comment by ghost** [comment]
there is one problem though, the surface can't officially boot from sd card so i think you have to like install to sd card and then like put the efi onto the internal ssd, also the efi badstorm made does detect sd cards fine for me

**Comment by albcal** [comment]
hi ! i'm trying to boot bigsur from sdcard, all the install process was successful but when i try to boot for first time stuck at 
Waiting for DSMOS...
ALF, old data swfs_pid_entry <ptr>, update rules_msg <ptr>, updaterules_state <ptr> 
I'm using BigSur 11.2.2   any help? thankyou

**Comment by albcal** [comment]
> nope. SD card can not be found in installer's disk util. But I can install to SD card using this efi `链接:https://pan.baidu.com/s/13sVRRqZySKvUu5saWpeOvQ 提取码:25i9`(from:https://www.bilibili.com/read/cv10042865)

Hi ! I'm triying to install bigsur and monterrey to sdcard without sucess, both can be installed to sdcard but in the final boot process something goes wrong, I'm using the EFI file you posted for bigsur and other for monterrey, wich macos you use ? Thanks

**Comment by badstorm** [comment]
I'm tring to install macOS on the SD with those steps:

- On Windows resize the main disk and leave 1GB of free space
- Install macOS in the SD with an USB adapter
- Create an APFS volume in the 1GB internal disk free space
- Move the `Preboot `partition form the SD to the internal disk
- Install OC in the EFI of the internal disk

But when I install macOS in the SD the first two steps of the installer works, but when the system boot before the finali install assistant i get KP and it reboot. This only appen when i boot from the SD (with usb adapter). With an USB disk works.

Do you have any idea why it not boot from SD?
